src/cli/tc.rs

```diff
@@ -60,51 +59,81 @@ pub fn run() -> Result<()> {
-    .prompt();
-
-  match answer {
-    Ok(true) => {
-      let file_name = inquire::Text::new("Enter the file name:")
-        .with_default(".graphql")
-        .prompt()
-        .unwrap_or_else(|_| String::from(".graphql"));
-
-      let file_name = format!("{}.graphql", file_name.strip_suffix(".graphql").unwrap_or(&file_name));
-
-      let confirm = Confirm::new(&format!("Do you want to create the file {}?", file_name))
-        .with_default(false)
-        .prompt();
-
-      match confirm {
-        Ok(true) => {
-          fs::write(format!("{}/{}", file_path, &file_name), "")?;
-
-          let graphqlrc = format!(
-            r#"|schema:
-               |- './{}'
-               |- './.tailcallrc.graphql'
-          "#,
-            &file_name
-          )
-          .strip_margin();
-          fs::write(format!("{}/.graphqlrc.yml", file_path), graphqlrc)?;
-        }
-        Ok(false) => (),
-        Err(e) => return Err(e.into()),
+  let folder_exists = fs::metadata(file_path).is_ok();
+
+  if !folder_exists {
+    let confirm = Confirm::new(&format!("Do you want to create the folder {}?", file_path))
+      .with_default(false)
+      .prompt();
+
+    match confirm {
+      Ok(true) => fs::create_dir(file_path)?,
+      Ok(false) => (),
+      Err(e) => return Err(e.into()),
+    };
+  }
+
+  let tailcallrc = include_str!("../../examples/.tailcallrc.graphql");
+
+  let file_name = ".tailcallrc.graphql";
+  let yml_file_name = ".graphqlrc.yml";
+  let yml_exists = fs::metadata(format!("{}/{}", file_path, yml_file_name)).is_ok();
+
+  if !yml_exists {
+    fs::write(format!("{}/{}", file_path, yml_file_name), "")?;
+
+    let graphqlrc = r#"|schema:
+         |- './.tailcallrc.graphql'
+    "#
+    .strip_margin();
+
+    fs::write(format!("{}/.graphqlrc.yml", file_path), graphqlrc)?;
+  }
+
+  let tailcall_exists = fs::metadata(format!("{}/{}", file_path, file_name)).is_ok();
+
+  if tailcall_exists {
+    // confirm overwrite
+    let confirm = Confirm::new(&format!("Do you want to overwrite the file {}?", file_name))
+      .with_default(false)
+      .prompt();
+
+    match confirm {
+      Ok(true) => fs::write(format!("{}/{}", file_path, file_name), tailcallrc.as_bytes())?,
+      Ok(false) => (),
+      Err(e) => return Err(e.into()),
+    };
+  } else {
+    fs::write(format!("{}/{}", file_path, file_name), tailcallrc.as_bytes())?;
+  }
+
+  let graphqlrc_path = format!("{}/.graphqlrc.yml", file_path);
+  let graphqlrc = fs::read_to_string(&graphqlrc_path)?;
+
+  if !graphqlrc.contains(file_name) {
+    let confirm = Confirm::new(&format!("Do you want to add {} to the schema?", file_name))
+      .with_default(false)
+      .prompt();
+
+    match confirm {
+      Ok(true) => {
+        let mut schema_line = graphqlrc
+          .lines()
+          .find(|line| line.contains("schema:"))
+          .unwrap()
+          .to_string();
+
+        schema_line.push_str("\n  - './.tailcallrc.graphql'");
```

Comment:
let's not play with raw strings here since it's too brittle. Let's utilize `serde_yaml` to work with yamls

Reply:
We've used raw strings before this change (and we're still using them when creating the file from scratch), but I can't find an example of manipulating `yaml` files; besides that, `serde_yaml` needs a struct to deserialize into a rust type – from my findings –, so I'm not sure if when serializing into a string again we'd have any loss in data

Reply:
what about emitting a message to the user to add `  - './.tailcallrc.graphql'` on their graphqlrc instead of us updating it?

Reply:
use this https://docs.rs/serde_yaml/latest/serde_yaml/value/index.html to parse any valid yaml

Reply:
Hey @neo773, as we're splitting this one, do you mind working on this part?

Reply:
@ologbonowiwi Didn't have write access so I committed here could you copy paste over to your brannch?
https://github.com/neo773/tailcall/commit/f1b8ceb90173f5fcca97ea5e5d9b69eb005592c2

Reply:
I'm having some issues to checkout into the branch locally here so I invited you to colaborate on my fork, can you push it?